dts/bindings/input/gpio-kbd-matrix.yaml

```
@@ -0,0 +1,40 @@
+# Copyright 2023 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  GPIO based keyboard matrix input device
+
+  Implement an input device for a GPIO based keyboard matrix.
+
+  Example configuration:
+
+  kbd-matrix {
+          compatible = "gpio-kbd-matrix";
+          row-gpios = <&gpio0 0 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>,
+                      <&gpio0 1 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+          col-gpios = <&gpio0 2 GPIO_ACTIVE_LOW>,
+                      <&gpio0 3 GPIO_ACTIVE_LOW>,
+                      <&gpio0 4 GPIO_ACTIVE_LOW>;
+          row-size = <2>;
+          col-size = <3>;
+          no-ghostkey-check;
+  };
+
+compatible: "gpio-kbd-matrix"
+
+include: kbd-matrix-common.yaml
```

```suggestion
          no-ghostkey-check;
  };

compatible: "gpio-kbd-matrix"

include: kbd-matrix-common.yaml
```

maybe filter-out row-size/col-size when including?

---

done, good idea, also done